### PR TITLE
Edge count optimization

### PR DESCRIPTION
**Pull request by DvirDukhan**
added typed edges count

### CODE REVIEW

File: src/execution_plan/optimizations/reduce_count.c [hunk 1]
```diff
@@ -144,6 +144,36 @@ static int _identifyEdgeCountPattern(OpBase *root, OpResult **opResult, OpAggreg
...
+    if(SINGLE_EDGE(*e)) {
+        *sum = *currentTotal + 1;
+    } else {
+        // Multiple edges
+        ids = (EdgeID*)(*e);
+        *sum = *currentTotal + array_len(ids);
+    }
+}
+
+uint64_t _countRelationshipEdges(GrB_Matrix relationshipMatrix){
```
Comment (swilly22) [inline src/execution_plan/optimizations/reduce_count.c]: Rename relationshipMatrix to `M`

File: src/execution_plan/optimizations/reduce_count.c [hunk 2]
```diff
@@ -144,6 +144,36 @@ static int _identifyEdgeCountPattern(OpBase *root, OpResult **opResult, OpAggreg
...
+    GrB_BinaryOp op;
+    GrB_Info info = GrB_BinaryOp_new(&op, _accumulateEdges, GrB_UINT64, GrB_UINT64, GrB_UINT64);
```
Comment (swilly22) [inline src/execution_plan/optimizations/reduce_count.c]: Rename _accumulateEdges to _countEdges

File: src/execution_plan/optimizations/reduce_count.c [hunk 3]
```diff
@@ -144,6 +144,36 @@ static int _identifyEdgeCountPattern(OpBase *root, OpResult **opResult, OpAggreg
...
+
+    GrB_Monoid monoid;
+    info = GrB_Monoid_new(&monoid, op, (uint64_t) 0);
```
Comment (swilly22) [inline src/execution_plan/optimizations/reduce_count.c]: global static init to NULL, created on first call.

File: src/execution_plan/optimizations/reduce_count.c [hunk 4]
```diff
@@ -144,6 +144,36 @@ static int _identifyEdgeCountPattern(OpBase *root, OpResult **opResult, OpAggreg
...
+        *sum = *currentTotal + 1;
+    } else {
+        // Multiple edges
+        ids = (EdgeID*)(*e);
+        *sum = *currentTotal + array_len(ids);
+    }
+}
+
+uint64_t _countRelationshipEdges(GrB_Matrix relationshipMatrix){
+    GrB_BinaryOp op;
```
Comment (swilly22) [inline src/execution_plan/optimizations/reduce_count.c]: Rename to a more meaningful name than op

File: src/execution_plan/optimizations/reduce_count.c [hunk 5]
```diff
@@ -144,6 +144,36 @@ static int _identifyEdgeCountPattern(OpBase *root, OpResult **opResult, OpAggreg
...
+    GrB_Info info = GrB_BinaryOp_new(&op, _accumulateEdges, GrB_UINT64, GrB_UINT64, GrB_UINT64);
+
+    GrB_Monoid monoid;
```
Comment (swilly22) [inline src/execution_plan/optimizations/reduce_count.c]: Rename to a more meaningful name than monoid.

File: src/execution_plan/optimizations/reduce_count.c (hunk 1 above)
Comment (DvirDukhan) [reply src/execution_plan/optimizations/reduce_count.c]: done

File: src/execution_plan/optimizations/reduce_count.c (hunk 2 above)
Comment (DvirDukhan) [reply src/execution_plan/optimizations/reduce_count.c]: done

File: src/execution_plan/optimizations/reduce_count.c (hunk 3 above)
Comment (DvirDukhan) [reply src/execution_plan/optimizations/reduce_count.c]: done

File: src/execution_plan/optimizations/reduce_count.c (hunk 4 above)
Comment (DvirDukhan) [reply src/execution_plan/optimizations/reduce_count.c]: done

File: src/execution_plan/optimizations/reduce_count.c (hunk 5 above)
Comment (DvirDukhan) [reply src/execution_plan/optimizations/reduce_count.c]: done

**Pull request merged**